config/routes.rb

```diff
@@ -0,0 +1,19 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'homepages#index'
+
+  resources :homepages, only: [:index]
+
+  resources :passengers do
+    resources :trips, only: [:index, :create, :update]
+  end
+
+  resources :drivers do
+    resources :trips, only: [:index, :assign_rating]
+  end
+
+  resources :drivers, except: [:show]
```

Comment:
You don't need to include `resources :drivers` twice. The `resources` method can take both a block and an `except` keyword argument.